argopy/tests/test_utilities.py

```diff
@@ -67,3 +68,25 @@ def test_clear_cache():
 #         assert isinstance(ds, xr.DataArray) is True
 #     except requests.HTTPError:  # not our fault
 #         pass
+
+
+def test_linear_interpolation_remap():
```

Comment:
We should test if the function throws errors when it should
So this would read:

```python
def test_linear_interpolation_remap():
    # create fake data to test interpolation:
    temp = np.random.rand(200, 100)
    pres = np.sort(np.floor(np.zeros(
        [200, 100])+np.linspace(50, 950, 100)+np.random.randint(-5, 5, [200, 100])))
    dsfake = xr.Dataset({'TEMP': (['N_PROF', 'N_LEVELS'], temp),
                         'PRES': (['N_PROF', 'N_LEVELS'],  pres)},
                        coords={'N_PROF': ('N_PROF', range(200)),
                                'N_LEVELS': ('N_LEVELS', range(100)),
                                'Z_LEVELS': ('Z_LEVELS', np.arange(100, 900, 20))})

    # Run it with success:
    dsi = linear_interpolation_remap(
        dsfake.PRES, dsfake['TEMP'], dsfake['Z_LEVELS'], z_dim='N_LEVELS', z_regridded_dim='Z_LEVELS')
    assert 'remapped' in dsi.dims

    # Test error:
    with pytest.raises(RuntimeError):  # catches error from _regular_interp linked to z_dim
        dsi = linear_interpolation_remap(
            dsfake.PRES, dsfake['TEMP'], dsfake['Z_LEVELS'], z_regridded_dim='Z_LEVELS')

    # Test error:
    with pytest.raises(ValueError):  # catches error from linear_interpolation_remap linked to datatype
        dsi = linear_interpolation_remap(
            dsfake.PRES, dsfake, dsfake['Z_LEVELS'], z_dim='N_LEVELS', z_regridded_dim='Z_LEVELS')
```

Reply:
or even better like this:

```python
class test_linear_interpolation_remap(unittest.TestCase):
    @pytest.fixture(autouse=True)
    def create_data(self):
        # create fake data to test interpolation:
        temp = np.random.rand(200, 100)
        pres = np.sort(np.floor(np.zeros(
            [200, 100])+np.linspace(50, 950, 100)+np.random.randint(-5, 5, [200, 100])))
        self.dsfake = xr.Dataset({'TEMP': (['N_PROF', 'N_LEVELS'], temp),
                             'PRES': (['N_PROF', 'N_LEVELS'],  pres)},
                            coords={'N_PROF': ('N_PROF', range(200)),
                                    'N_LEVELS': ('N_LEVELS', range(100)),
                                    'Z_LEVELS': ('Z_LEVELS', np.arange(100, 900, 20))})

    def test_interpolation(self):
        # Run it with success:
        dsi = linear_interpolation_remap(
            self.dsfake.PRES, self.dsfake['TEMP'], self.dsfake['Z_LEVELS'], z_dim='N_LEVELS', z_regridded_dim='Z_LEVELS')
        assert 'remapped' in dsi.dims

    def test_error_zdim(self):
        # Test error:
        with pytest.raises(RuntimeError):  # catches error from _regular_interp linked to z_dim
            dsi = linear_interpolation_remap(
                self.dsfake.PRES, self.dsfake['TEMP'], self.dsfake['Z_LEVELS'], z_regridded_dim='Z_LEVELS')

    def test_error_ds(self):
        # Test error:
        with pytest.raises(ValueError):  # catches error from linear_interpolation_remap linked to datatype
            dsi = linear_interpolation_remap(
                self.dsfake.PRES, self.dsfake, self.dsfake['Z_LEVELS'], z_dim='N_LEVELS', z_regridded_dim='Z_LEVELS')
```